src/Command/MigrationsCommand.php

```diff
@@ -61,23 +72,25 @@ class MigrationsShell extends Shell
      */
     public function getOptionParser(): ConsoleOptionParser
     {
-        return parent::getOptionParser()
-            ->addOption('plugin', ['short' => 'p'])
-            ->addOption('target', ['short' => 't'])
-            ->addOption('connection', ['short' => 'c'])
-            ->addOption('source', ['short' => 's'])
-            ->addOption('seed')
-            ->addOption('ansi')
-            ->addOption('no-ansi')
-            ->addOption('no-lock', ['boolean' => true])
-            ->addOption('force', ['boolean' => true])
-            ->addOption('version', ['short' => 'V'])
-            ->addOption('no-interaction', ['short' => 'n'])
-            ->addOption('template', ['short' => 't'])
-            ->addOption('format', ['short' => 'f'])
-            ->addOption('only', ['short' => 'o'])
-            ->addOption('dry-run', ['short' => 'x'])
-            ->addOption('exclude', ['short' => 'e']);
+        if (parent::defaultName() === 'migrations') {
+            return parent::getOptionParser();
+        }
+        $parser = parent::getOptionParser();
+        $command = new MigrationsDispatcher::$phinxCommands[static::$commandName]();
+        $parser->setDescription($command->getDescription());
+        $definition = $command->getDefinition();
+        foreach ($definition->getOptions() as $key => $option) {
+            if (!empty($option->getShortcut())) {
+                $parser->addOption($option->getName(), [
+                    'short' => $option->getShortcut(),
+                    'help' => $option->getDescription(),
```

Comment:
Could options with short cuts have default values or be boolean flags?

Reply:
yes they have `boolean` value but does not matter 
because these options are only for `CompetionCommand` and all other commands 
even `-h` /  `--help` parse and executes by symfony not cake , so we dont need to these values , we just need to know options name 

see my commit about this 
https://github.com/cakephp/migrations/pull/399/commits/86092ad70